src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs

```diff
@@ -8069,6 +7607,35 @@ public static IEnumerable<object[]> JsonEncodedTextStrings
                 };
             }
         }
+
+        private static IEnumerable<JsonWriterOptions> JsonOptions()
+        {
+            return from indented in new[] { true, false }
+                   from skipValidation in new[] { true, false }
+                   from indentCharacter in indented ? new char?[] { null, ' ', '\t' } : []
+                   from indentSize in indented ? new int?[] { null, 0, 1, 2, 127 } : []
+                   select CreateOptions(indented, indentCharacter, indentSize, skipValidation);
+
+            static JsonWriterOptions CreateOptions(bool indented, char? indentCharacter, int? indentSize, bool skipValidation)
+            {
+                var options = new JsonWriterOptions { Indented = indented, SkipValidation = skipValidation };
+                if (indentCharacter is not null) options.IndentCharacter = (char)indentCharacter;
+                if (indentSize is not null) options.IndentSize = (int)indentSize;
+                return options;
+            }
+        }
+
+        private static IEnumerable<object[]> JsonOptionsWith<T>(IEnumerable<T> others) =>
+            from options in JsonOptions()
+            from inputValue in others
+            select new object[] { options, inputValue };
+
+        private static IEnumerable<object[]> JsonOptionsWith<T, U>(IEnumerable<T> others, IEnumerable<U> anothers) =>
+            from options in JsonOptions()
+            from inputValue in others
+            from anotherValue in anothers
+            select new object[] { options, inputValue, anotherValue };
+
     }
```

Comment:
Do we have any tests validating that indentation isn't being produced if we only set an indentation character and/or indentation length without setting the `Indented` property?

Reply:
Test added.